packages/transactions/lib/createMarketSellTransaction.ts

```diff
@@ -0,0 +1,82 @@
+import _ from 'lodash'
+import { getAmmAccount } from '@play-money/accounts/lib/getAmmAccount'
+import { getUserAccount } from '@play-money/accounts/lib/getUserAccount'
+import { costToHitProbability, sell } from '@play-money/amms/lib/maniswap-v1'
+import { createTransaction, TransactionItemInput } from './createTransaction'
+import { convertMarketSharesToPrimary, convertPrimaryToMarketShares } from './exchanger'
+
+type MarketSellTransactionInput = {
+  userId: string
+  amount: number // in shares
+  sellCurrencyCode: 'YES' | 'NO'
+  marketId: string
+}
+
+export async function createMarketSellTransaction({
+  userId,
+  marketId,
+  amount,
+  sellCurrencyCode,
+}: MarketSellTransactionInput) {
+  const userAccount = await getUserAccount({ id: userId })
+  const ammAccount = await getAmmAccount({ marketId })
+
+  // When selling shares, the number of shares will decrease some by filling amm/limit orders.
+  // We need an equilivant number of yes and no shares to get money back out of the exchanger.
+  let outstandingShares = amount
+  let oppositeOutstandingShares = 0
+  let oppositeCurrencyCode = sellCurrencyCode === 'YES' ? 'NO' : 'YES'
+  let accumulatedTransactionItems: Array<TransactionItemInput> = []
+  // To account for floating point errors, we will limit the number of loops to a sane number.
+  let maximumSaneLoops = 100
+
+  while (outstandingShares !== oppositeOutstandingShares && maximumSaneLoops > 0) {
```

Comment:
Maybe we should check for outstandingShares and oppositeOutstandingShares to be between some epsilon of each other, so that it doesn't do 100 useless loops (and create a bunch of useless transaction items)?

This shouldn't matter once we switch to fixed point decimals though.